lib/resources/users.js

```diff
@@ -101,36 +130,27 @@ module.exports = (service, endpoint) => {
     Users.getByActorId(params.id)
       .then(getOrNotFound)
       .then((user) => auth.canOrReject('user.update', user.actor)
-        .then(() => User.fromApi(body))
+        .then(() => {
+          if (oidc.isEnabled()) {
+            // don't allow modifying email or password for users when using OIDC
+            // REVIEW is there some official, lower-level place to do this, e.g. in lib/model/frames.js?
+            const { email, password, ...filtered } = body;
+            return filtered;
+          } else {
+            return body;
+          }
+        })
+        .then((filteredBody) => User.fromApi(filteredBody))
```

Comment:
> is there some official, lower-level place to do this, e.g. in lib/model/frames.js?

Yes, this is usually handled by the frame, specifically its `fromApi()` method. By default, the `fromApi()` method only accepts fields that have been marked `writable` on the frame (or on an aux frame). That means that `password` should be automatically filtered out because it's not a `writable` field.

`email` is a tougher case because we want to filter it out only sometimes. I think it's reasonable to do that here in the endpoint rather than the frame. An additional complication is that the release criteria now call for admins to be able to update emails.

Altogether, I'm thinking this could look something like:

```js
Users.getByActorId(params.id)
  .then(getOrNotFound)
  .then((user) => auth.canOrReject('user.update', user.actor)
    .then(() => (oidc.isEnabled() ? auth.can('user.update', User.species) : resolve(true))
      .then((canUpdateEmail) => (canUpdateEmail ? body : without(['email'], body))))
    .then((filteredBody) => User.fromApi(filteredBody))
```

Reply:
Also, you can see that some user endpoints use async/await rather than promise chains. You should feel free to convert this endpoint to async/await if that makes things easier!

Reply:
Comment updated.